### PR TITLE
Re-haul Elixir documentation

### PR DESCRIPTION
**Pull request by whatyouhide**
## Pre-merge checklist

*If you work at Sentry, you're able to merge your own PR without review, but please don't unless there's a good reason.*

- [ ] Checked Vercel preview for correctness, including links
- [ ] PR was reviewed and approved by any necessary SMEs
- [ ] PR was reviewed and approved by a member of the [Sentry docs team](https://github.com/orgs/getsentry/teams/docs)

## Description of changes

Woah, this is a big PR. This brings the Elixir docs in line with other SDKs. It essentially removes all the Elixir-specific stuff and takes advantage of everything in `common`. Exciting.

### CODE REVIEW

**Comment by whatyouhide** [comment]
Rebased!

File: src/platform-includes/set-release/elixir.mdx [hunk 1]
```diff
@@ -1,4 +1,4 @@
 ```elixir {config/prod.exs}
```
Comment (shanamatthews) [inline src/platform-includes/set-release/elixir.mdx:1]: Need to fix syntax so filename works

`elixir {filename:config/prod.exs}`

**Review by shanamatthews — COMMENTED** [review]
Wow, great job! These docs look great to me, just a few small suggestions.

One other thing... Any idea why the `PlatformSdkDetail` component isn't there for the Elixir SDK?

**Comment by shanamatthews** [comment]
Oh and don't worry about the linter error, I can quickly fix that before merge.

File: src/platform-includes/capture-error/elixir.mdx [hunk 2]
```diff
@@ -0,0 +1,10 @@
+In Elixir, you can use the `:logger` handler (`Sentry.LoggerHandler`) to automatically capture all logged exception (such as the ones from crashing processes). Alternatively, you can use `Sentry.capture_exception/2` to manually capture and report exceptions:
```
[the remark this answers is not in the document]
Comment (whatyouhide) [reply src/platform-includes/capture-error/elixir.mdx:1]: Yes!

File: src/platform-includes/getting-started-config/elixir.mdx [hunk 3]
```diff
@@ -0,0 +1,13 @@
+Sentry has a range of configuration options, but most applications will have a configuration that looks like the following:
+
+<SignInNote />
+
+```elixir
```
[the remark this answers is not in the document]
Comment (whatyouhide) [reply src/platform-includes/getting-started-config/elixir.mdx:5]: Yes absolutely, good catch!

File: src/platform-includes/configuration/config-intro/elixir.mdx [hunk 4]
```diff
@@ -0,0 +1,12 @@
+In Elixir, configuration is handled using the standard Elixir configuration ([`Config`](https://hexdocs.pm/elixir/Config.html)).
+
+Add configuration to the `:sentry` application in the your config file.
+
+<SignInNote />
+
+```elixir
```
[the remark this answers is not in the document]
Comment (whatyouhide) [reply src/platform-includes/configuration/config-intro/elixir.mdx:7]: Yessss!

File: src/platform-includes/configuration/before-send/elixir.mdx
[the remark this answers is not in the document]
Comment (whatyouhide) [reply src/platform-includes/configuration/before-send/elixir.mdx:1]: Yes, done.

File: src/platform-includes/getting-started-install/elixir.mdx [hunk 5]
```diff
@@ -0,0 +1,14 @@
+Edit your `mix.exs` file to add it as a dependency and add the `:sentry` package to your application:
+
+```elixir {filename:mix.exs}
+defp deps do
+  [
+    # ...
+    {:sentry, "~> 9.0"},
```
[the remark this answers is not in the document]
Comment (whatyouhide) [reply src/platform-includes/getting-started-install/elixir.mdx:7]: Awesome, yes, TIL!

**Comment by whatyouhide** [comment]
I addressed all the (great) feedback 🙃 

> One other thing... Any idea why the PlatformSdkDetail component isn't there for the Elixir SDK?

Nope. Is this something I define in this repo? If so, where?

**Review by sl0thentr0py — APPROVED** [review]
tysm @whatyouhide <3

**Comment by sl0thentr0py** [comment]
@shanamatthews the `PlatformSdkDetail` stuff has to do with the registry, [the elixir package metadata](https://github.com/getsentry/sentry-release-registry/blob/master/packages/hex/sentry/9.1.0.json) is there but I think there's something else missing but I don't know what exactly.

**Comment by shanamatthews** [comment]
Thanks @sl0thentr0py. No reason to block this based on the registry data missing in that case.

**Pull request merged**